src/client/mesh.cpp

```diff
@@ -202,6 +203,20 @@ void setMeshColor(scene::IMesh *mesh, const video::SColor &color)
 		setMeshBufferColor(mesh->getMeshBuffer(j), color);
 }
 
+template <typename F>
+static void applyToMesh(scene::IMesh *mesh, const F &fn)
```

Comment:
Why is the template needed here?
`const F &fn` seems to be `const std::function<void(video::S3DVertex *vertex)> &fn` everywhere I looked.

Reply:
@Quent42340 Lambda’s type is *not* `std::function<whatever>`, it’s some internal type, different for each lambda. I agree that that’s all *castable* to `std::function`, but would that be inlined then? That’s critical as the loop is tight.

Reply:
@numberZero I know, but I really think this kind of cast is trivial.